include/SH3/system/glbuffer.hpp

```diff
@@ -0,0 +1,112 @@
+/** @file
+ *
+ *  A wrapper and more OO approach to OpenGL Vertex Buffer Objects. This is also helps prevent any
+ *  memory problems as well as providing a whole lot of helper functions for our VBO.
+ *
+ *  @copyright 2017 Palm Studios
+ *
+ *  @date 8-6-2017
+ *
+ *  @author Jesse Buhagiar
+ */
+#ifndef GLBUFFER_HPP_INCLUDED
+#define GLBUFFER_HPP_INCLUDED
+
+#include "GL/glew.h"
+#include "GL/gl.h"
+
+#include <string>
+#include <cstddef>
+
+namespace sh3_gl
+{
+    /**
+     *
+     * An OpenGL Vertex Buffer Object (VBO). Provides a nice, easy, typesafe way
+     * to create and manage OpenGL buffers.
+     */
+    struct sh3_glbuffer final
+    {
+    public:
+
+        enum Target
+        {
+            SH3_ARRAY_BUFFER    = GL_ARRAY_BUFFER,
+            SH3_ELEMENT_BUFFER  = GL_ELEMENT_ARRAY_BUFFER,
+            SH3_TEXTURE_BUFFER  = GL_TEXTURE_BUFFER
+        };
+
+        /**
+         *  Default constructor.
+         */
+        sh3_glbuffer(){Create();buffType = Target::SH3_ARRAY_BUFFER;};
+        sh3_glbuffer(Target type){Create();buffType = type;};
+        sh3_glbuffer(Target type, const std::string& buffname){Create();buffType = type, name = buffname;};
+
+        /**
+         *  Destructor. Ensure we have no leaks when this buffer is to be deleted by the C++ Runtime.
+         */
+        ~sh3_glbuffer(){/**Release();*/};
```

Comment:
I'm not quite sure how OpenGL works on this instance. Release frees the buffer's ID, and I'm not entirely sure if the data is released with it. If we create the buffers locally in say `sh3_model::Load( )`, and then release is called when the function returns, the data should be deleted from the GPU (and all the buffers in buffer becoming invalid)? I'll need to a bit more investigation, hence the commenting. I'll remove the line completely for now.